packages/react-components/react-combobox/src/components/Dropdown/useButtonTriggerSlot.ts

```diff
@@ -104,9 +104,12 @@ export function useButtonTriggerSlot(
         searchString.current = '';
       }, 500);
 
+      if (open) {
+        moveToNextMatchingOptionWithSameCharacterHandling();
+      }
```

Comment:
This change is because of the difference in the activedescendant being set when the listbox is rendered, before being opened. The first press of a letter should only open the listbox, and not also move to the second option starting with that letter.